vulnfeeds/cmd/cvelist2osv/main.go

```diff
@@ -59,26 +65,419 @@ func extractConversionMetrics(cve cves.CVE5, refs []osvschema.Reference) {
+
+	// Handle cases where a product is marked as "affected" by default, and specific versions are marked "unaffected".
+	if affected.DefaultStatus == "affected" {
+		// For Linux kernel CVEs, this logic is often handled by CPEs, so we skip it here.
+		if cnaAssigner == "Linux" {
+
+			notes = append(notes, "Skipping Linux Affected range versions in favour of CPE versions")
+			return nil, "", notes
+		}
+		// Calculate the affected ranges by finding the inverse of the unaffected ranges.
+		ranges, inverseNotes := findInverseAffectedRanges(affected)
+		notes = append(notes, inverseNotes...)
+
+		if len(ranges) != 0 {
+			return ranges, string(osvschema.RangeSemVer), notes
+		}
+		return nil, "", notes
+	}
+
+	versionTypesCount := make(map[string]int)
+
+	for _, vers := range affected.Versions {
+		if vers.Status != "affected" {
+			continue
+		}
+		if _, ok := versionTypesCount[vers.VersionType]; ok {
+			versionTypesCount[vers.VersionType]++
+		} else {
+			versionTypesCount[vers.VersionType] = 0
+		}
+
+		var introduced, fixed, lastaffected string
+
+		// Quality check the version strings to avoid using filler content.
+		vQuality := vulns.CheckQuality(vers.Version)
+		if vQuality >= vulns.Filler {
+			notes = append(notes, fmt.Sprintf("Version value for %s %s is filler or empty", affected.Vendor, affected.Product))
+		}
+		vLessThanQual := vulns.CheckQuality(vers.LessThan)
+		vLTOEQual := vulns.CheckQuality(vers.LessThanOrEqual)
+
+		hasRange := vLessThanQual <= vulns.Spaces || vLTOEQual <= vulns.Spaces
+		notes = append(notes, fmt.Sprintf("Range detected: %v", hasRange))
+		// Handle cases where 'lessThan' is mistakenly the same as 'version'.
+		if vers.LessThan != "" && vers.LessThan == vers.Version {
+			notes = append(notes, fmt.Sprintf("Warning: lessThan (%s) is the same as introduced (%s)\n", vers.LessThan, vers.Version))
+			hasRange = false
+		}
+
+		if hasRange {
+			if vQuality <= vulns.Spaces {
+				introduced = vers.Version
+				notes = append(notes, fmt.Sprintf("%s - Introduced from version value - %s", vQuality.String(), vers.Version))
+			}
+			if vLessThanQual <= vulns.Spaces {
+				fixed = vers.LessThan
+				notes = append(notes, fmt.Sprintf("%s - Fixed from LessThan value - %s", vLessThanQual.String(), vers.LessThan))
+			} else if vLTOEQual <= vulns.Spaces {
+				lastaffected = vers.LessThanOrEqual
+				notes = append(notes, fmt.Sprintf("%s - LastAffected from LessThanOrEqual value- %s", vLTOEQual.String(), vers.LessThanOrEqual))
+			}
+
+			if introduced != "" && fixed != "" {
+				versionRange := osvschema.Range{}
+				addToVersionRange(&versionRange, introduced, "", fixed)
+				versionRanges = append(versionRanges, versionRange)
+			} else if introduced != "" && lastaffected != "" {
+				versionRange := osvschema.Range{}
+				addToVersionRange(&versionRange, introduced, lastaffected, "")
+				versionRanges = append(versionRanges, versionRange)
+			}
+		} else {
```

Comment:
If I'm reading the nesting correctly, you can flatten this else statement and just end the hasRange part with a continue.